### PR TITLE
add M4 as build dep for GCC 5.1.0

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
@hajgato: please test on the same system you were testing #1694 on?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 3 (3 easyconfigs in this PR)
Linux RHEL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/92d84d9319a900a07409 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
Linux RHEL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/1d557e1274865fa7e87e for a full test report.

File: easybuild/easyconfigs/m/M4/M4-1.4.17.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+easyblock = 'ConfigureMake'
+
+name = 'M4'
+version = '1.4.17'
+
+homepage = 'http://www.gnu.org/software/m4/m4.html'
+description = """GNU M4 is an implementation of the traditional Unix macro processor.
+It is mostly SVR4 compatible although it has some extensions 
+(for example, handling more than 9 positional parameters to macros).
+GNU M4 also has built-in functions for including files, running shell commands, doing arithmetic, etc.
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/m/M4/M4-1.4.17.eb]: LGTM, just 1-space indent the above.

**Comment by hajgato** [comment]
Test report by @hajgato
**FAILED**
Build succeeded for 3 out of 5 (3 easyconfigs in this PR)
Linux opensuse 13.1, Intel(R) Xeon(R) CPU           W3530  @ 2.80GHz, Python 2.7.6
See https://gist.github.com/249041f6d18627a9c87e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
Linux RHEL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/b1cc19c0884fc15c0446 for a full test report.

**Comment by hajgato** [comment]
Test report by @hajgato
**SUCCESS**
Build succeeded for 5 out of 5 (3 easyconfigs in this PR)
Linux opensuse 13.1, Intel(R) Xeon(R) CPU           W3530  @ 2.80GHz, Python 2.7.6
See https://gist.github.com/97d623bff1c54c1e4f3b for a full test report.

**Comment by wpoely86** [comment]
looks fine

**Comment by boegel** [comment]
Good to go, thanks for testing @hajgato and reviewing @wpoely86!

**Pull request merged**